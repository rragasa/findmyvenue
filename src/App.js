import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchVenues } from './utils/actions/appActions';
import Typography from '@material-ui/core/Typography';
import Search from './components/Search';
import Venues from './components/Venues';
import {
  apiVersion as v,
  client_id,
  client_secret,
} from './config.json';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(location) {
    const { getVenues } = this.props;
    const payload = {
      v,
      client_id,
      client_secret,
      near: location,
    };

    getVenues(payload);
  }

  render() {
    const { venues = {} } = this.props;
    return (
      <div className="App">
       <header className="App-header">
       <Typography variant="h3" component="h3">Find my venue</Typography>
      </header>
      <Search onSubmit={(location) => this.handleSubmit(location)}/>
      { venues.length > 0 && (
        <Venues venues={venues}/>
      )}
      </div>
     );
  }
}
const mapStateToProps = ({
  appReducer: {
    venues,
  },
 }) => ({
   venues,
});

 const mapDispatchToProps = dispatch => ({
  getVenues: (payload) => dispatch(fetchVenues(payload))
 })

 export default connect(mapStateToProps, mapDispatchToProps)(App);
