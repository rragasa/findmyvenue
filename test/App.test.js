import App from '../src/App';

describe('App Component', () => {
  it('should render the app component', () => {
      let wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
  });
});