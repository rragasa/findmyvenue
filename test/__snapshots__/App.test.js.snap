// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the app component 1`] = `ShallowWrapper {}`;
